src/views/darkroom.c

```diff
@@ -3872,7 +3872,18 @@ static gboolean _second_window_delete_callback(GtkWidget *widget,
                                                GdkEvent *event,
                                                dt_develop_t *dev)
 {
-  _darkroom_ui_second_window_write_config(dev->second_wnd);
```

Comment:
@TurboGit I think your problem is here. When the window is closed, GTK will trigger an event. In the cleanup function, it sets `dev->second_wnd = NULL;`. When the event handler is called, I think `_darkroom_ui_second_window_write_config` is invoked with nullptr. Maybe something in this function doesn't like it when you use nullptr.

I can't confirm if this is the case since I don't have a linux machine at the moment.